mirrord/layer/src/socket/ops.rs

```diff
@@ -631,3 +634,47 @@ pub(super) fn getaddrinfo(
 
     Detour::Success(result)
 }
+
+fn remote_read_into_string(path: &str, read_length: usize) -> Detour<CString> {
+    let hostname_path = CString::new(path)?;
+
+    let hostname_fd = file::ops::open(
+        Some(&hostname_path),
+        OpenOptionsInternal {
+            read: true,
+            ..Default::default()
+        },
+    )?;
+
+    let hostname_file = file::ops::fgets(hostname_fd, read_length)?;
+
+    close_layer_fd(hostname_fd);
+
+    Detour::Success(CString::new(
+        hostname_file
+            .bytes
+            .into_iter()
+            .take(hostname_file.read_amount as usize)
+            .collect::<Vec<_>>(),
+    )?)
+}
+
+/// Resolve the fake local address to the real local address.
+#[tracing::instrument(level = "trace", skip(raw_name, name_length))]
+pub(super) fn gethostname(raw_name: *mut c_char, name_length: usize) -> Detour<i32> {
```

Comment:
we prefer `ops` logic to be safe, so I suggest that `gethostname` returns `Detour<String>` then the caller does the actual assignment